packages/worker/src/replay.ts

```diff
@@ -33,25 +33,18 @@ export class ReplayError extends Error {
   }
 }
 
-export interface ReplayResults {
-  readonly hasErrors: boolean;
-  /** Maps run id to information about the replay failure */
-  readonly errors: ReplayError[];
-}
-
-/**
- * @internal
- */
-export interface ReplayRunOptions {
-  failFast?: boolean;
+export interface ReplayResult {
+  workflowId: string;
+  runId: string;
+  error?: ReplayError;
 }
 
 /**
  * An iterable on worflow histories and their IDs, used for batch replaying.
  *
  * @experimental - this API is not considered stable
  */
-export type ReplayHistoriesIterable = AsyncIterable<HistoryAndWorkflowId> | Iterable<HistoryAndWorkflowId>;
+export type ReplayHistoriesIterable = AsyncIterableIterator<HistoryAndWorkflowId> | Iterable<HistoryAndWorkflowId>;
```

Comment:
Why impose `AsyncIterableIterator` here? Also, it should be coherent with the non-sync type.

Reply:
I wanted it to have a `next` awaitable so I could race it the run promise but ended up not using that.
I think it's fine to revert.

Reply:
On second thought let's keep the change in case we need to await `next`. I can see it come in handy when Core supports concurrent replays.